File: ai-engine/app/routers/weather.py
from fastapi import APIRouter, HTTPException, Query
from app.pipelines.weather import get_weather

router = APIRouter()


@router.get("/current")
async def current_weather(lat: float = Query(...), lng: float = Query(...)):
    try:
        return await get_weather(lat, lng)
    except Exception as e:
        raise HTTPException(502, f"Weather service unavailable: {e}")


@router.get("/route-advice")
async def route_advice(lat: float = Query(...), lng: float = Query(...)):
    try:
        weather = await get_weather(lat, lng)
        return {
            "condition": weather["condition"],
            "advice": weather["navigationAdvice"],
            "alerts": weather["alerts"],
            "isHazardous": weather["isHazardous"],
        }
    except Exception as e:
        raise HTTPException(502, f"Weather service unavailable: {e}")

File: backend/src/routes/weather.ts
import type { FastifyInstance } from "fastify";
import { z } from "zod";
import { aiClient } from "../services/aiClient";

export async function weatherRoutes(app: FastifyInstance) {
  app.get("/current", async (req, reply) => {
    const { lat, lng } = z.object({
      lat: z.coerce.number(),
      lng: z.coerce.number(),
    }).parse(req.query);
    try {
      const data = await aiClient.getWeather(lat, lng);
      return data;
    } catch {
      return reply.status(502).send({ error: "Weather service unavailable" });
    }
  });

  app.get("/route-advice", async (req, reply) => {
    const { lat, lng } = z.object({
      lat: z.coerce.number(),
      lng: z.coerce.number(),
    }).parse(req.query);
    try {
      const data = await aiClient.getWeatherAdvice(lat, lng);
      return data;
    } catch {
      return reply.status(502).send({ error: "Weather service unavailable" });
    }
  });
}
